### DISCUSSION TITLE
DreamPI support

### DISCUSSION THREAD

**Discussion by sairuk** [retronas/discussions#36] [General]
Discussion around DreamPi support.

DreamPi requires additional hardware to get the Dreamcast Modem hooked, but the right hardware could have additional uses for other retro hardware. Feel free to throw some options down or let us know if you have something setup etc.

https://dreamcast.wiki/DreamPi

Modem options
- USB Voice Modem _(officially listed)_

Perhaps with a little bit of hackery, these could work
- USB TTL device to RJ11
- Serial to RJ11 adapter combined with TCPSer

**Comment by danmons** [comment]
Definitely wanted.  I've got this in [IDEAS.md](https://github.com/danmons/retronas/blob/main/IDEAS.md), but keen to flesh out the idea more here.

DreamCast is the obvious use case because that's what it was designed for.  But there's no reason it shouldn't work with GameCube's dialup adaptor too (for any GC fans reading, it's unlikely that it will work with FSP and ISO loading, this would be for online gaming only).

This video from Youtuber Mac84 shows that a Clamshell iBook running (I assume) OS9 can use DreamPi to get online.  The video also shows the 9V injection needed for North America (apparently Australia and Europe need 2x9V=18V?) but that can all be worked out in testing:
https://youtu.be/NFUTInM7gq8

And then from there tools like [WebOne](https://github.com/danmons/retronas/wiki/WebOne
) (already implemented and working via TCP from any browser that supports standard HTTP Proxying) are there to act as a proxy for older browsers.  In theory that should make anything else work - Win3.X + Trumpet Windsock, Win9X native, Amiga, etc.

I had a USB modem on order from AliExpress, but checking just the other day that one got cancelled by the vendor. I'll order a couple more from around the place, as it's worth having different ones to test anyway.

**Reply by sairuk** [reply]
I am going to chase a couple as well, should be a lot of fun to get going

**Comment by danmons** [comment]
Noted in here: https://github.com/danmons/retronas/discussions/76

DreamPi forces DNS to go out to an external provider (I forget whether it's 1.1.1.1 or 8.8.8.8, but it's one of them. I checked in the source code about a week ago).

Would be good if we do ever implement DreamPi to consider dhcp/dns ramifications, and if needed change that to the local network's DHCP provided DNS server via values sent to RetroNAS and relayed along via dnsmasq.

**Reply by sairuk** [reply]
dnsmasq was next on my list to post something about because of this as well, see you beat me too it. Great stuff

**Comment by sairuk** [comment]
We've implemented dreampi 'as-is' support the intent being to turn the RetroNAS into as close to a dreampi as possible. The support is a culmination of work by others on the python3 port. 

You may also use dreampi with a Sega Saturn according to some posts I have seen.

I have successfully played quake3 online (dc) with a fresh RetroNAS install on a 2B (install is slow) with a diy vli.

See:
* https://github.com/danmons/retronas/wiki/DreamPi
* https://github.com/danmons/retronas/issues/274
